### ISSUE TITLE
ABORT RTC::Producer::ActivateStreamProfiles() | failed assertion (profile already in active profiles map)

### ISSUE THREAD

**Issue by ibc** [versatica/mediasoup#235]
### Your environment

- mediasoup version: 2.5.5

### Issue description

Fuzzing RTP (still to know what it consisted of, we'll know it soon) a worker died:

```
2018-12-11T11:16:09.753Z mediasoup-demo-server:INFO:Room mediaPeer "newproducer" event [peer.name:ohcdknre, id:46893738]
mediasoup-worker's stderr: ABORT[id:kyfydczr#2] RTC::Producer::ActivateStreamProfiles() | failed assertion `this->mapActiveProfiles.find(profile) == this->mapActiveProfiles.end()': profile already in active profiles map
```

**Comment by ibc** [comment]
The scenario seems to be this:

* Firefox sending RTX, but mediasoup is not yet ready for it (nor mediasoup-client nor the RID mechanism in mediasoup).
* RTX streams have the same RID as the media stream they protect to.
* I think that in mediasoup we are assuming that it's a new media stream but then we inspect the RID and it matches a previous one, so something crashes because we assume that such a profile should not be present.

**Comment by ibc** [comment]
So here the problem:

* A non announced RTX video packet with an unknown SSRC but a known RID arrives to `Producer::MayNeedNewStream()`.
* It does not find such a new SSRC in any map so it inspects its RID value.
* It finds an `encoding` in `RtpParameters` with that RID (`encodingId`), so it calls `CreateRtpStream(encoding, ssrc)`.
* In `CreateRtpStream()` it does not find any RTX stuff since it was not announced by the client.
* `CreateRtpStream()` does:
```c++
// Add the stream to the profiles map.
this->mapRtpStreamProfiles[rtpStream].insert(profile);
```
* Finally `CreateRtpStream()` calls `ActivateStreamProfiles(rtpStream)`.
* And `ActivateStreamProfiles()` does the assertion that crashes:
```c+
auto& profiles = this->mapRtpStreamProfiles[rtpStream];

// Notify about the profiles being enabled.
for (auto& profile : profiles)
{
	MS_ASSERT(
	  this->mapActiveProfiles.find(profile) == this->mapActiveProfiles.end(),
	  "profile already in active profiles map");
```

**Comment by ibc** [comment]
This bug basically means that, if the browser sends a new SSRC with same RID, it will crash, no matter it's RTX or a fake stream.

**Comment by ibc** [comment]
I'm working on this and get some issues:

* What are we doing [here](https://github.com/versatica/mediasoup/blob/2.5.5/worker/src/RTC/Producer.cpp#L556)? If this is a RTX packet, are we also creating a `RtpStream` and storing it in the maps???

**Comment by ibc** [comment]
Fixed in #238
